diff -urN icewm-1.0.6.org/Makefile.in icewm-1.0.6/Makefile.in
--- icewm-1.0.6.org/Makefile.in	Wed Jan 17 20:34:20 2001
+++ icewm-1.0.6/Makefile.in	Wed Jan 17 20:35:33 2001
@@ -67,37 +67,37 @@
 # Makefile TABS *SUCK*
 install-base: base
 	@echo ------------------------------------------
-	@echo Installing icewm to $(BINDIR)
-	@$(INSTALLDIR) $(BINDIR)
+	@echo Installing icewm to $(DESTDIR)$(BINDIR)
+	@$(INSTALLDIR) $(DESTDIR)$(BINDIR)
 	@for app in $(BINFILES) ; do \
-            $(INSTALLBIN) $$app $(BINDIR); \
+            $(INSTALLBIN) $$app $(DESTDIR)$(BINDIR); \
         done
-	@echo Installing defaults, icons and themes to $(LIBDIR)
-	@$(INSTALLDIR) $(LIBDIR)
-	@$(INSTALLDIR) $(CFGDIR)
+	@echo Installing defaults, icons and themes to $(DESTDIR)$(LIBDIR)
+	@$(INSTALLDIR) $(DESTDIR)$(LIBDIR)
+	@$(INSTALLDIR) $(DESTDIR)$(CFGDIR)
 	@for lib in $(LIBFILES) ; do \
-            $(INSTALLLIB) $$lib $(LIBDIR); \
+            $(INSTALLLIB) $$lib $(DESTDIR)$(LIBDIR); \
         done
 	@for xpmdir in $(XPMDIRS) ; do \
-            $(INSTALLDIR) $(LIBDIR)/$$xpmdir; \
+            $(INSTALLDIR) $(DESTDIR)$(LIBDIR)/$$xpmdir; \
             for icon in lib/$$xpmdir/*.xpm ; do \
-                $(INSTALLLIB) $$icon $(LIBDIR)/$$xpmdir; \
+                $(INSTALLLIB) $$icon $(DESTDIR)$(LIBDIR)/$$xpmdir; \
             done; \
         done
 	@for theme in $(THEMES) ; do \
             echo Installing theme: $$theme; \
-            $(INSTALLDIR) $(LIBDIR)/themes/$$theme; \
+            $(INSTALLDIR) $(DESTDIR)$(LIBDIR)/themes/$$theme; \
             for pixmap in lib/themes/$$theme/*.xpm ; do \
-                $(INSTALLLIB) $$pixmap $(LIBDIR)/themes/$$theme; \
+                $(INSTALLLIB) $$pixmap $(DESTDIR)$(LIBDIR)/themes/$$theme; \
             done; \
             for config in lib/themes/$$theme/*.theme ; do \
-                $(INSTALLLIB) $$config $(LIBDIR)/themes/$$theme; \
+                $(INSTALLLIB) $$config $(DESTDIR)$(LIBDIR)/themes/$$theme; \
             done; \
             for l in $(XPMDIRS) ; do \
                 if [ -d lib/themes/$$theme/$$l ] ; then \
-                    $(INSTALLDIR) $(LIBDIR)/themes/$$theme/$$l; \
+                    $(INSTALLDIR) $(DESTDIR)$(LIBDIR)/themes/$$theme/$$l; \
                     for f in lib/themes/$$theme/$$l/* ; do \
-                        $(INSTALLLIB) $$f $(LIBDIR)/themes/$$theme/$$l; \
+                        $(INSTALLLIB) $$f $(DESTDIR)$(LIBDIR)/themes/$$theme/$$l; \
                     done;\
                 fi;\
             done;\
@@ -106,24 +106,24 @@
 
 install-docs: docs $(LSM)
 	@echo ------------------------------------------
-	@$(INSTALLDIR) $(DOCDIR)/icewm-$(VERSION)
-	cp $(DOCFILES) $(DOCDIR)/icewm-$(VERSION)
-	cp -r doc $(DOCDIR)/icewm-$(VERSION)
+	@$(INSTALLDIR) $(DESTDIR)$(DOCDIR)/icewm-$(VERSION)
+	cp $(DOCFILES) $(DESTDIR)$(DOCDIR)/icewm-$(VERSION)
+	cp -r doc $(DESTDIR)$(DOCDIR)/icewm-$(VERSION)
 
 install-nls: nls
 	@echo ------------------------------------------
-	@cd po ; ${MAKE} install
+	@cd po ; ${MAKE} DESTDIR=$(DESTDIR) install
 
 install-guievents: guievents
 	@echo ------------------------------------------
 	@echo Installing GUI event support to $(BINDIR)
-	@$(INSTALLDIR) $(BINDIR)
+	@$(INSTALLDIR) $(DESTDIR)$(BINDIR)
 	@for a in $(GUIEVENTSFILES) ; do \
-            $(INSTALLBIN) $$a $(BINDIR); \
+            $(INSTALLBIN) $$a $(DESTDIR)$(BINDIR); \
         done
 
 install-gnome: gnome
 	@echo ------------------------------------------
-	@echo Installing icewm-gnome to $(WMPROPDIR)
-	@$(INSTALLDIR) $(WMPROPDIR)
-	@$(INSTALLLIB) lib/IceWM.desktop $(WMPROPDIR)
+	@echo Installing icewm-gnome to $(DESTDIR)$(WMPROPDIR)
+	@$(INSTALLDIR) $(DESTDIR)$(WMPROPDIR)
+	@$(INSTALLLIB) lib/IceWM.desktop $(DESTDIR)$(WMPROPDIR)
diff -urN icewm-1.0.6.org/configure.in icewm-1.0.6/configure.in
--- icewm-1.0.6.org/configure.in	Wed Jan 17 20:34:20 2001
+++ icewm-1.0.6/configure.in	Wed Jan 17 20:34:27 2001
@@ -66,7 +66,7 @@
 AC_FUNC_SELECT_ARGTYPES
 
 dnl Configuring features
-TARGETS='base docs'
+TARGETS='base'
 
 dnl Checking for X11
 AC_PATH_XTRA
diff -urN icewm-1.0.6.org/po/Makefile icewm-1.0.6/po/Makefile
--- icewm-1.0.6.org/po/Makefile	Wed Jan 17 20:34:20 2001
+++ icewm-1.0.6/po/Makefile	Wed Jan 17 20:34:27 2001
@@ -23,8 +23,8 @@
 	  LANG=`echo $$c|sed -e 's/.mo//'` ; \
 	  MSGDIR="$(LOCDIR)/$$LANG/LC_MESSAGES"; \
 	  echo Installing language: $$LANG ; \
-	  $(INSTALLDIR) $$MSGDIR; \
-	  $(INSTALLBIN) $$c $$MSGDIR/$(PACKAGE).mo; \
+	  $(INSTALLDIR) $(DESTDIR)$$MSGDIR; \
+	  $(INSTALLBIN) $$c $(DESTDIR)$$MSGDIR/$(PACKAGE).mo; \
 	done
 
 clean:
