--- icewm-1.2.25.orig/src/ylocale.cc	2006-02-03 07:24:27.000000000 +0100
+++ icewm-1.2.25/src/ylocale.cc	2006-07-25 16:27:42.000000000 +0200
@@ -152,7 +152,7 @@
     char * inbuf((char *) lStr), * outbuf((char *) uStr);
     size_t inlen(lLen), outlen(4 * lLen);
 
-    if (0 > (int) iconv(instance->toUnicode, const_cast<const char **>(&inbuf), &inlen, &outbuf, &outlen))
+    if (0 > (int) iconv(instance->toUnicode, &inbuf, &inlen, &outbuf, &outlen))
         warn(_("Invalid multibyte string \"%s\": %s"), lStr, strerror(errno));
 
     *((YUChar *) outbuf) = 0;
