--- icewm-1.3.12/configure.ac.orig	2020-11-11 11:21:19.688467839 +0100
+++ icewm-1.3.12/configure.ac	2020-11-11 11:23:25.721118395 +0100
@@ -347,7 +347,7 @@
 AC_ARG_ENABLE([menus-gnome2],
     AC_HELP_STRING([--disable-menus-gnome2],[Disable display of GNOME 2 menus.]))
 if test x$enable_menus_gnome2 != xno; then
-    PKG_CHECK_MODULES([GNOME2],[gnome-desktop-2.0 libgnomeui-2.0],[
+    PKG_CHECK_MODULES([GNOME2],[gnome-desktop-2.0 gnome-vfs-2.0 libgnomeui-2.0],[
 	GNOME_VER=2
 	AC_DEFINE([CONFIG_GNOME_MENUS],[1],[Define to make IceWM more GNOME-friendly])
 	BUILD_MENU_GNOME2=yes
Index: icewm-1.3.7/src/yapp.cc
===================================================================
--- icewm-1.3.7.orig/src/yapp.cc	2010-10-31 15:09:36.000000000 +0100
+++ icewm-1.3.7/src/yapp.cc	2011-03-14 15:58:59.000000000 +0100
@@ -269,7 +269,7 @@
         }
 #endif
 
-        {
+        if(measure_latency) {
             struct timeval difftime;
             struct timeval curtime;
             gettimeofday(&curtime, 0);
