diff -ruN icewm-1.2.2_/src/wmprog.cc icewm-1.2.2/src/wmprog.cc
--- icewm-1.2.2_/src/wmprog.cc	Fri Sep  6 19:18:07 2002
+++ icewm-1.2.2/src/wmprog.cc	Sat Sep 28 18:15:04 2002
@@ -586,7 +586,9 @@
         YIcon *gnomeIcon = 0;
         YIcon *kdeIcon = 0;
 
-        char *gnomeAppsMenu = gnome_datadir_file("gnome/apps/");
+// kde menu don't seem to work
+//        char *gnomeAppsMenu = gnome_datadir_file("gnome/apps/");
+        char *gnomeAppsMenu = gnome_datadir_file("applnk");
         char *gnomeUserMenu = gnome_util_home_file("apps/");
         const char *kdeMenu = strJoin(kdeDataDir, "/applnk", 0);
 
