diff -urN aaa/icewm-1.2.3pre1/src/aapm.cc icewm-1.2.3pre1/src/aapm.cc
--- aaa/icewm-1.2.3pre1/src/aapm.cc	Sun Oct 20 16:04:55 2002
+++ icewm-1.2.3pre1/src/aapm.cc	Fri Nov 15 01:12:20 2002
@@ -463,7 +463,7 @@
 
     if (gradient) {
         g.copyPixbuf(*gradient, this->x(), this->y(),
-                     width((), height(), 0, 0));
+                     width(), height(), 0, 0);
     }
     else
 #endif
diff -urN aaa/icewm-1.2.3pre1/src/icesound.cc icewm-1.2.3pre1/src/icesound.cc
--- aaa/icewm-1.2.3pre1/src/icesound.cc	Sun Oct 20 16:04:55 2002
+++ icewm-1.2.3pre1/src/icesound.cc	Fri Nov 15 01:13:20 2002
@@ -82,6 +82,10 @@
 #define OSS_DEFAULT_DEVICE "/dev/dsp"
 #define YIFF_DEFAULT_SERVER "127.0.0.1:9433"
 
+#define THROW(Result) { rc = (Result); goto exceptionHandler; }
+#define TRY(Command) { if ((rc = (Command))) THROW(rc); }
+#define CATCH(Handler) { exceptionHandler: { Handler } return rc; }
+
 /******************************************************************************
  * IceSound application
  ******************************************************************************/
