diff -ruN icewm-1.2.10pre1./src/default.h icewm-1.2.10pre1/src/default.h
--- icewm-1.2.10pre1./src/default.h	2003-06-29 19:44:56.000000000 +0200
+++ icewm-1.2.10pre1/src/default.h	2003-07-02 20:48:01.000000000 +0200
@@ -406,7 +406,7 @@
 #endif
     OBV("ConfirmLogout",                        &confirmLogout,                 "Confirm logout"),
 #ifdef CONFIG_SHAPED_DECORATION
-    OBV("ShapesProtectClientWindow",            &protectClientWindow,           "Don't cut client windows by shapes set trough frame corner pixmap")
+    OBV("ShapesProtectClientWindow",            &protectClientWindow,           "Don't cut client windows by shapes set trough frame corner pixmap"),
 #endif
     OIV("BorderSizeX",                          &wsBorderX, 0, 128,             "Horizontal window border"),
     OIV("BorderSizeY",                          &wsBorderY, 0, 128,             "Vertical window border"),
