diff -ruN icewm-1.2.15pre1./src/base.h icewm-1.2.15pre1/src/base.h
--- icewm-1.2.15pre1./src/base.h	2004-07-06 22:05:46.914841864 +0200
+++ icewm-1.2.15pre1/src/base.h	2004-07-06 22:05:32.211077176 +0200
@@ -171,7 +171,6 @@
 int strpcmp(char const *str, char const *pfx, char const *delim = "=:");
 unsigned strtoken(const char *str, const char *delim = " \t");
 char const * strnxt(const char *str, const char *delim = " \t");
-extern "C" char *basename(const char *filename);
 
 bool strequal(const char *a, const char *b);
 int strnullcmp(const char *a, const char *b);
diff -ruN icewm-1.2.15pre1./src/icesound.cc icewm-1.2.15pre1/src/icesound.cc
--- icewm-1.2.15pre1./src/icesound.cc	2004-07-06 22:07:53.000000000 +0200
+++ icewm-1.2.15pre1/src/icesound.cc	2004-07-06 22:12:24.544392984 +0200
@@ -190,7 +190,8 @@
     virtual int init(int & argc, char **& argv);
     
 private:
-    friend class CommandLine : public YCommandLine {
+    friend class CommandLine;
+    class CommandLine : public YCommandLine {
     public:
 	CommandLine(int & argc, char **& argv, YOSSAudio & oss):
 	    YCommandLine(argc, argv), oss(oss) {}
@@ -326,7 +327,8 @@
     virtual int init(int & argc, char **& argv);
 
 private:
-    friend class CommandLine : public YCommandLine {
+    friend class CommandLine;
+    class CommandLine : public YCommandLine {
     public:
 	CommandLine(int & argc, char **& argv, YESDAudio & esd):
 	    YCommandLine(argc, argv), esd(esd) {}
@@ -502,7 +504,8 @@
     	      Coefficient lVol = 1.0, Coefficient rVol = 1.0);
 
 private:
-    friend class CommandLine : public YCommandLine {
+    friend class CommandLine;
+    class CommandLine : public YCommandLine {
     public:
 	CommandLine(int & argc, char **& argv, YY2Audio & yiff):
 	    YCommandLine(argc, argv), yiff(yiff) {}
